sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/FormRecognizerAsyncClient.java

```diff
@@ -512,12 +514,17 @@
         };
     }
 
-    private Function<PollingContext<OperationResult>, Mono<List<RecognizedForm>>>
-        fetchExtractReceiptResult(boolean includeFieldElements) {
+    /*
+     * Poller's FETCHING operation that fetch a list of RecognizedForm.
+     */
+    private Function<PollingContext<OperationResult>, Mono<List<RecognizedForm>>> fetchingOperation(
+        Function<UUID, Mono<SimpleResponse<AnalyzeOperationResult>>> fetchingFunction,
+        boolean includeFieldElements) {
+
         return (pollingContext) -> {
             try {
                 final UUID resultUid = UUID.fromString(pollingContext.getLatestResponse().getValue().getResultId());
-                return service.getAnalyzeReceiptResultWithResponseAsync(resultUid)
+                return fetchingFunction.apply(resultUid)
                     .map(modelSimpleResponse -> {
                         throwIfAnalyzeStatusInvalid(modelSimpleResponse.getValue());
                         return toRecognizedForm(modelSimpleResponse.getValue().getAnalyzeResult(),
```

Comment:
To make this method common to all the API's we could have it return the `modelSimpleResponse` and let the respective API's do their conversion `toRecognizedForm()` or `toFormPage`?